src/pytest_flyte/plugin.py

```diff
@@ -123,20 +149,57 @@ def docker_cleanup():
 
 
 @pytest.fixture(scope="session")
-def flyteclient(docker_ip, docker_services, docker_compose, capsys_suspender):
-    port = docker_services.port_for("backend", 30081)
-    url = f"{docker_ip}:{port}"
-    os.environ["FLYTE_PLATFORM_URL"] = url
-    os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
-
-    def _check():
-        try:
-            docker_compose.execute("exec backend wait-for-flyte.sh")
-            return True
-        except Exception as e:
-            print(e)
-            return False
-
-    with capsys_suspender():
-        docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
-    return friendly.SynchronousFlyteClient(url, insecure=True)
+def flyteclient(request):
+    if not request.config.getoption("--proto-path"):
+        raise ValueError("Serialized Data Proto Path must be set")
+
+    if request.config.getoption("--local") in ["True", "true", True]:
+        docker_ip = request.getfixturevalue("docker_ip")
+        docker_services = request.getfixturevalue("docker_services")
+        docker_compose = request.getfixturevalue("docker_compose")
+        capsys_suspender = request.getfixturevalue("capsys_suspender")
+
+        port = docker_services.port_for("backend", 30081)
+        url = f"{docker_ip}:{port}"
+        os.environ["FLYTE_PLATFORM_URL"] = url
+        os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
+
+        # create flytectl config
```

Comment:
Let's commit the sandbox flytectl config in the boilerplate and reuse the same config everywhere in our test.